### ISSUE TITLE
[BUG] npm version command issuing invalid git command to check repository status

### ISSUE THREAD

**Issue by bcowgill** [npm/cli#2871]
### Current Behavior:
I am running the npm version command but it is issuing an invalid git command to check whether the repository is clean and therefore always fails.

### Expected Behavior:
npm should either check the git version or issue an alternate --porcelain syntax to work properly with git, or report which version of git it requires to operate correctly.

### Steps To Reproduce:
1. In the environment shown below
2. Run 'npm version patch -m "release Version patch %s, now with double cleaning action." '
3. See error:
npm ERR! code 129
npm ERR! command failed
npm ERR! command /usr/bin/git status --porcelain=v1 -uno
npm ERR! error: option `porcelain' takes no value
npm ERR! usage: git status [options] [--] <pathspec>...
npm ERR! 
npm ERR!     -v, --verbose         be verbose
npm ERR!     -s, --short           show status concisely
npm ERR!     -b, --branch          show branch information
npm ERR!     --porcelain           machine-readable output

### Environment:
- OS: Ubuntu 14.04
- Node: 12.19.0
- npm: 7.5.6
- git: 1.9.1

Note, when I run the command /usr/bin/git status --porcelain -uno
instead, it correctly shows the status of the repo so it seems npm is expecting some version of git without verifying that it has that version.

**Comment by ljharb** [comment]
git v1.9 is very ancient; it seems reasonable that npm ensures that git is a version released more recently than ~6 years ago.

**Comment by bcowgill** [comment]
Yep, I realise this, perhaps at least give a warning message about what version of git you support.  

For now I'm modifying my pre-version scripts to do the right thing as I cannot easily upgrade git on this machine.  

An apparently innocuous npm install -g npm broke me. :-(

I guess it's time to upgrade to a new laptop ;-)

**Comment by ljharb** [comment]
Given the number of unpatched security vulnerabilities in v1.9, if you really can't upgrade git, it might be :-)

**Comment by darcyclarke** [comment]
Resolution here on our end is to provide more clear docs/messaging for unsupported versions of deps we have (like `git`)
